modules/aws-eks-managed-node-groups/main.tf

```diff
@@ -82,3 +82,17 @@ resource "aws_eks_node_group" "managed_ng" {
   ]
 
 }
+
+# NOTE: Tags need to replicated manually to the ASGs as per this issue: https://github.com/aws/containers-roadmap/issues/608
+# We won't need to have this resource when managed node groups in EKS replicate tags to ASGs
+resource "aws_autoscaling_group_tag" "cluster_autoscaler_tag" {
+  for_each = local.asg_tags
+
+  autoscaling_group_name = aws_eks_node_group.managed_ng.resources[0].autoscaling_groups[0].name
```

Comment:
this can be problematic, we should avoid embedding this into the module - see https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1558#issuecomment-1030640207

Reply:
@bryantbiggs what's the alternative then? I was able to create the cluster without any problems. The dependency issue was when I was trying to add a tag with the cluster id which you get it after the cluster is created. But the tags I left in the example work. To avoid this problem, this section could be **only** for the tags I'd use for CA to scale-down to zero nodes, and not leave it open to include any other tags. What do you think?

Reply:
the guidance for tags on EKS managed node groups is:
1. Use a custom launch template which will propagate tags down to the ASGs and instances
2. If using default launch template provided by EKS managed node group, then users can use this `aws_autoscaling_group_tag` resource in their implementation

Reply:
> when we fix the tagging problem this hack will go away
Correct - which is why its better to externalize it. If we bake it into the module and remove it later, its a breaking change. If users use the tag resource and native support is offered through the API in the future, its a quality of life enhancement

Reply:
so perhaps it makes sense to add another example that illustrates both scenarios? otherwise, this will make it existing examples way more complex ... I have customers asking me for an example on this one